Fix rich media issues (multiple 3D models, video looping and media pausing)

LucasLacerdaUX opened the pull request:
**Why are these changes introduced?**
Fixes #674 

**What approach did you take?**
- For the multiple model-viewers, I've fixed a typo on `global.js` where it was missing the `model` object when trying to access `modelViewerUI`.
- For video looping, I used an invalid variable on `product-media.liquid`. Reverted `loop` like the other video types. 
- For auto pausing, I've moved the `window.pauseAllMedia()` to `modal-dialog`'s `show()` function. I've opted to do it there instead of on the context of `product-modal` because this pattern makes sense on every modal dialog (it's possible to have a video inside of a popup block).

**Demo links**

- [Store](https://os2-demo.myshopify.com/?preview_theme_id=126428872726)
- [Editor](https://os2-demo.myshopify.com/admin/themes/126428872726/editor)

**Testing**
- [ ] Go to a [product page with multiple 3D models](https://os2-demo.myshopify.com/products/test-3d-models-video?preview_theme_id=126428872726). Try playing and interacting with multiple 3D models on the page.
- [ ] Go to a [product page with HTML5 videos](https://os2-demo.myshopify.com/products/test-3d-models-video?preview_theme_id=126428872726), set `Enable video looping` setting to true and check if it works inside the modal.
- [ ] Go to a [product page with videos](https://os2-demo.myshopify.com/products/test-3d-models-video?preview_theme_id=126428872726) on desktop. Play a video outside the modal. Click on an image to open the modal. The video should automatically pause.

**Checklist**
- [x] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [x] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [x] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [x] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [x] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

ludoboludo reviewed (approved):
Tested and worked as expected 👌 

Also if you play a video (not in the modal), then click on an image to open the modal, it won't pause the video. 

I wonder if it's something that could be tackled here too 🤔

LucasLacerdaUX commented:
Good catch, @ludoboludo. I've updated the PR to fix this issue. 

I've opted to do it in `modal-dialog` context instead of `product-modal` because this pattern makes sense on every modal dialog (it's possible to have a video inside of a popup block). I've also removed a redundant media pause functionality from `product-modal` - it's already there on its parent custom element class.

LucasLacerdaUX commented on assets/global.js:
```diff
@@ -454,6 +454,7 @@ class ModalDialog extends HTMLElement {
     this.setAttribute('open', '');
     if (popup) popup.loadContent();
     trapFocus(this, this.querySelector('[role="dialog"]'));
+    window.pauseAllMedia();
```
Good point.

My assumption was that modals would **always** cover the content on the screen, so that's why it wouldn't make sense to keep media in the background being played. Because elements outside the modal are not accessible (focus trap + background overlay) unless you dismiss it, I couldn't think of a scenario where pausing this content wouldn't be desired. But yeah, there might be.

I do agree with you that there's opportunity to make this class more generic and this is a really good point to validate in a refactor. For the uses we have now, I think it's okay to make it part of the general `ModalDialog` element.
